cli/tests/unit/command_test.ts

```diff
@@ -0,0 +1,505 @@
+  async function runRedirectStdin() {
+    const tempDir = await Deno.makeTempDir();
+    const fileName = tempDir + "/redirected_stdio.txt";
+    const encoder = new TextEncoder();
+    await Deno.writeFile(fileName, encoder.encode("hello"));
+    const file = await Deno.open(fileName);
+    const fileReader = readableStreamFromReader(file);
+
+    const cmd = new Deno.Command(Deno.execPath(), {
+      args: [
+        "eval",
+        "if (new TextDecoder().decode(await Deno.readAll(Deno.stdin)) !== 'hello') throw new Error('Expected \\'hello\\'')",
+      ],
+    });
+    const child = cmd.spawn({
+      stdin: "piped",
+    });
+    await fileReader.pipeTo(child.stdin);
+
+    const status = await child.wait();
+    assertEquals(status.code, 0);
+  },
+);
+
+Deno.test(
+  { permissions: { run: true, read: true } },
+  async function runEnv() {
+    const cmd = new Deno.Command(Deno.execPath(), {
+      args: [
+        "eval",
+        "Deno.stdout.write(new TextEncoder().encode(Deno.env.get('FOO') + Deno.env.get('BAR')))",
+      ],
+      env: {
+        FOO: "0123",
+        BAR: "4567",
+      },
+    });
+
+    const output = await cmd.output();
+    const s = new TextDecoder().decode(output.stdout);
+    assertEquals(s, "01234567");
+  },
+);
+
+Deno.test(
+  { permissions: { run: true, read: true } },
+  async function killSuccess() {
+    const cmd = new Deno.Command(Deno.execPath(), {
+      args: ["eval", "setTimeout(() => {}, 10000)"],
+    });
+    const child = cmd.spawn();
+
+    child.kill("SIGKILL");
+    const status = await child.wait();
+
+    assertEquals(status.success, false);
+    if (Deno.build.os === "windows") {
+      assertEquals(status.code, 1);
+      assertEquals(status.signal, undefined);
+    } else {
+      assertEquals(status.code, 137);
+      assertEquals(status.signal, 9);
+    }
+  },
+);
+
+Deno.test(
+  { permissions: { run: true, read: true } },
+  async function killFailed() {
+    const cmd = new Deno.Command(Deno.execPath(), {
+      args: ["eval", "setTimeout(() => {}, 5000)"],
+    });
+    const child = cmd.spawn();
+
+    assertThrows(() => {
+      // @ts-expect-error testing runtime error of bad signal
+      child.kill("foobar");
+    }, TypeError);
+
+    await child.wait();
+  },
+);
+
+Deno.test(
+  { permissions: { run: true, read: true, env: true } },
+  async function clearEnv(): Promise<void> {
+    const cmd = new Deno.Command(Deno.execPath(), {
+      args: [
+        "eval",
+        "-p",
+        "JSON.stringify(Deno.env.toObject())",
+      ],
+      clearEnv: true,
+      env: {
+        FOO: "23147",
+      },
+    });
+
+    const output = await cmd.output();
+    const obj = JSON.parse(new TextDecoder().decode(output.stdout));
+
+    // can't check for object equality because the OS may set additional env vars for processes
+    // so we check if PATH isn't present as that is a common env var across OS's and isn't set for processes.
```

Comment:
Long lines. The linter is lenient on long comments but, _ceteris paribus_, prefer wrapping at 80 columns.

Reply:
actually just copied from `cli/tests/unit/process_test.ts`, but sure, will fix both